lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,944 @@
+# Copyright (c) 2017-2018 Dell EMC Inc.
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import os
+import json
+import re
+import xml.etree.ElementTree as ET
+from distutils.version import LooseVersion
+from datetime import datetime
+from ansible.module_utils.urls import open_url
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+
+HEADERS = {'content-type': 'application/json'}
+
+
+class RedfishUtils(object):
+
+    def __init__(self, creds, root_uri):
+        self.root_uri = root_uri
+        self.creds = creds
+        self._init_session()
+        self.default_system_id = None
+        self.default_chassis_id = None
+        self.default_manager_id = None
+        return
+
+    # The following functions are to send GET/POST/PATCH/DELETE requests
+    def get_request(self, uri):
+        if 'token' in self.creds:
+            headers = {"X-Auth-Token": self.creds['token']}
+        try:
+            resp = open_url(uri, method="GET",
+                            url_username=self.creds['user'],
+                            url_password=self.creds['pswd'],
+                            force_basic_auth=True, validate_certs=False,
+                            timeout=10, use_proxy=False)
+            data = json.loads(resp.read())
+        except HTTPError as e:
+            return {'ret': False, 'msg': "HTTP Error: %s" % e.code}
+        except URLError as e:
+            return {'ret': False, 'msg': "URL Error: %s" % e.reason}
+        # Almost all errors should be caught above, but just in case
+        except:
+            return {'ret': False, 'msg': "Unknown error"}
+        return {'ret': True, 'data': data}
+
+    def post_request(self, uri, pyld, hdrs):
+        if 'token' in self.creds:
+            headers = {"X-Auth-Token": self.creds['token']}
```

Comment:
Does this variable get consumed anywhere?

Reply:
Not in this current implementation. Not opposed to removing it.

Reply:
It's fine if it's a placeholder for future capabilities.